File: implementations/node/sdk/src/contracts/observation/execution-step-observation.ts
import type { AiSdkExecutionStepStatus } from "./execution-step-status.js";

export interface AiSdkExecutionStepObservation {
  readonly name: string;
  readonly stepKey: string;
  readonly status: AiSdkExecutionStepStatus;
  readonly startedAtUtc?: string;
  readonly updatedAtUtc?: string;
  readonly completedAtUtc?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/observation/execution_step_observation.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from .execution_step_status import AiSdkExecutionStepStatus


@dataclass(frozen=True)
class AiSdkExecutionStepObservation(AiSdkWireModel):
    name: str
    step_key: str
    status: AiSdkExecutionStepStatus
    started_at_utc: str | None = None
    updated_at_utc: str | None = None
    completed_at_utc: str | None = None
